silkworm/core/execution/evm.cpp

```diff
@@ -203,7 +204,7 @@ evmc::Result EVM::call(const evmc_message& message) noexcept {
     evmc::Result res{EVMC_SUCCESS, message.gas};
 
     const auto value{intx::be::load<intx::uint256>(message.value)};
-    if (message.kind != EVMC_DELEGATECALL && state_.get_balance(message.sender) < value) {
+    if (!gas_bailout_ && message.kind != EVMC_DELEGATECALL && state_.get_balance(message.sender) < value) {
         res.status_code = EVMC_INSUFFICIENT_BALANCE;
         return res;
     }
```

Comment:
I would move the condition on `gas_bailout` inside the existing condition to emphasise its role by also adding a comment.

Moreover, I suppose we need the same additional check on gas bailout also in `EVM::create`.

Reply:
Left in this way as discussed